--- libmowgli-2-libmowgli-2.0.0/src/libmowgli/vio/vio_openssl.c~	2012-04-18 12:56:14.000000000 +0200
+++ libmowgli-2-libmowgli-2.0.0/src/libmowgli/vio/vio_openssl.c	2018-09-25 00:43:57.628671319 +0200
@@ -134,23 +134,7 @@
 	mowgli_ssl_connection_t *connection = vio->privdata;
 	const SSL_METHOD *method;
 
-	switch (connection->settings.ssl_version)
-	{
-	case MOWGLI_VIO_SSLFLAGS_SSLV2:
-		method = SSLv23_server_method();
-		break;
-	case MOWGLI_VIO_SSLFLAGS_SSLV3:
-		method = SSLv3_server_method();
-		break;
-	case MOWGLI_VIO_SSLFLAGS_TLSV10:
-	case MOWGLI_VIO_SSLFLAGS_TLSV11:
-	case MOWGLI_VIO_SSLFLAGS_TLSV12:
-		method = TLSv1_server_method();
-		break;
-	default:
-		/* Compat method */
-		method = SSLv23_server_method();
-	}
+	method = TLS_server_method();
 	
 	connection->ssl_context = SSL_CTX_new((SSL_METHOD *)method);
 	if (connection->ssl_context == NULL)
@@ -266,23 +250,7 @@
 
 	vio->error.op = MOWGLI_VIO_ERR_OP_CONNECT;
 
-	switch (connection->settings.ssl_version)
-	{
-	case MOWGLI_VIO_SSLFLAGS_SSLV2:
-		method = SSLv23_client_method();
-		break;
-	case MOWGLI_VIO_SSLFLAGS_SSLV3:
-		method = SSLv3_client_method();
-		break;
-	case MOWGLI_VIO_SSLFLAGS_TLSV10:
-	case MOWGLI_VIO_SSLFLAGS_TLSV11:
-	case MOWGLI_VIO_SSLFLAGS_TLSV12:
-		method = TLSv1_client_method();
-		break;
-	default:
-		/* Compat method */
-		method = SSLv23_client_method();
-	}
+	method = TLS_client_method();
 
 	/* Cast is to eliminate an excessively bogus warning on old OpenSSL --Elizacat */
 	connection->ssl_context = SSL_CTX_new((SSL_METHOD *)method);
